docs/site/migration/models/methods.md

```diff
@@ -6,8 +6,485 @@ sidebar: lb4_sidebar
+override this method in your repository class.
+
+The following LB4 code snippets show how to customize the **find** method of
+`NoteRepository`.
+
+**src/repositories/note.repository.ts**
+
+```ts
+export class NoteRepository extends DefaultCrudRepository<
+  Note,
+  typeof Note.prototype.id,
+  NoteRelations
+> {
+  constructor(
+    @inject('datasources.db') dataSource: DbDataSource,
+    @inject('my.cache') private cache: Map<string, Note[]>,
+  ) {
+    super(Note, dataSource);
+  }
+
+  async find(
+    filter?: Filter<Note>,
+    options?: Options,
+  ): Promise<(Note & NoteRelations)[]> {
+    let key: string = '';
+    if (filter) {
+      key = JSON.stringify(filter);
+    }
+
+    let cachedResults: Note[] | undefined = this.cache.get(key);
+
+    if (cachedResults) {
+      console.log('serving from cache');
+      return cachedResults;
+    } else {
+      console.log('serving from db');
+
+      let results: Note[] | undefined = await super.find(filter, options);
+      this.cache.set(key, results);
+
+      return results;
+    }
+  }
+}
+```
+
+The cache is injected into the repository.
+
+It was defined in the application class:
+
+**src/application.ts**
+
+```ts
+export class NoteApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  private cache: Map<string, Note[]> = new Map();
+
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    this.bind('my.cache').to(this.cache);
+
+    // ...
+  }
+}
+```
+
+No changes are made to any REST API endpoint settings in `NoteController`.
+
+## Add a New Model Method And a New Endpoint
+
+LoopBack 3 and 4 provide a lot of out-of-the-box functionality for developers
+with regards to data-access APIs and REST APIs. Only a few steps are required to
+create minor additions.
+
+### LoopBack 3 Approach
+
+The following LB3 code snippet shows how to customize the `Note` model by adding
+a **new** persisted model method `findByTitle`, and defining a **new** remote
+method for its respective endpoint `/findByTitle`. The full endpoint url will
+end up being `/Notes/findByTitle`.
+
+**common/models/Note.js**
+
+```js
+module.exports = function(Note) {
+  Note.remoteMethod('findByTitle', {
+    http: {
+      path: '/findByTitle',
+      verb: 'get',
+    },
+    accepts: {arg: 'title', type: 'string'},
+    returns: {arg: 'note', type: [Note], root: true},
+  });
+
+  Note.findByTitle = function(title, cb) {
+    var titleFilter = {
+      where: {
+        title: title,
+      },
+    };
+    Note.find(titleFilter, function(err, foundNotes) {
+      if (err) {
+        cb(err);
+      } else {
+        cb(null, foundNotes);
+      }
+    });
+  };
+};
+```
+
+### LoopBack 4 Approach
+
+To accomplish the same thing in LoopBack 4, we can add a **new** method
+`findByTitle` to `NoteRepository`, a **new** method `findByTitle` to
+`NoteController`, and specify an endpoint url of `/notes/byTitle/{title}`.
+
+**src/repositories/note.repository.ts**
```

Comment:
Can you update to use `{% include code-caption.html content="..." %}` here and similar areas?